Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [3]:
import numpy as np
X=[100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [4]:
## cреднее арифметическое
x_shtrih=sum(X)/len(X)
x_shtrih

65.3

In [5]:
## среднее квадратичное отклонение
q=[]
for x in X:
    qq=(x-x_shtrih)**2
    q.append(qq)
S=sum(q)/(len(X))

np.sqrt(S)

30.823854398825596

In [6]:
## Дисперссия смещенная и не Смещенная
q=[]
for x in X:
    qq=(x-x_shtrih)**2
    q.append(qq)
D_displ=sum(q)/len(X)
D_not_displ=sum(q)/(len(X)-1)

print ('Смещенная =',D_displ)
print ('Не смещенная =',D_not_displ)

Смещенная = 950.11
Не смещенная = 1000.1157894736842


В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Решение:
    
Достать 3 белых можно 3мя способами:

1) 0 из первого и 3 из второго 
2) 1 из первого и 2 из второго
3) 2 из первого и 1 из второго

In [15]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [23]:
## вероятность достать 0 черных из первого ящика и 3 из второго
v1=combinations(3,2)*combinations(5,0)/combinations(8,2)*(combinations(5,3)*combinations(7,1)/combinations(12,4))
v1

0.01515151515151515

In [24]:
## вероятность достать 1 белых из первого и 2 из второго
v2=combinations(5,1)*combinations(3,1)/combinations(8,2)*(combinations(5,2)*combinations(7,2)/combinations(12,4))
v2

0.22727272727272727

In [25]:
## вероятность достать 2 белых из первого и 1 из второго
v3=combinations(5,2)*combinations(3,0)/combinations(8,2)*(combinations(5,1)*combinations(7,3)/combinations(12,4))
v3

0.12626262626262627

In [26]:
## вероятность что 3 меча будут белые
v=v1+v2+v3
v

0.3686868686868687

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом

Решение:
Для решения используем формулу Байеса

$$P(B\:|\:A) = \frac{P(B) \cdot P(A\:|\:B)}{P(A)}$$

$$P(A\:|\:B_1) = 0.9$$

$$P(A\:|\:B_2) = 0.8$$

$$P(A\:|\:B_3) = 0.6$$


Поддставяляя соответсвующие значения найдем вероятности для спортсменов:

а)
$$P(B_1\:|\:A) = \frac{P(B_1) \cdot P(A\:|\:B_1)}{P(A)}= \frac{\frac{1}{3} \cdot 0.9}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} \approx 0.39$$

б)
$$P(B_2\:|\:A) = \frac{P(B_2) \cdot P(A\:|\:B_2)}{P(A)}= \frac{\frac{1}{3} \cdot 0.8}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} \approx 0.34$$

в)
$$P(B_3\:|\:A) = \frac{P(B_3) \cdot P(A\:|\:B_3)}{P(A)}= \frac{\frac{1}{3} \cdot 0.6}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} \approx 0.26$$

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Решение:

Для решения используем формулу Байеса

$$P(B\:|\:A) = \frac{P(B) \cdot P(A\:|\:B)}{P(A)}$$

Обозначим B_1 - студенты фактултета A, B_2 - студенты факультета B, B_3 - студенты факультета С. Тогда

$$P(A\:|\:B_1) = 0.8$$

$$P(A\:|\:B_2) = 0.7$$

$$P(A\:|\:B_3) = 0.9$$
    
Поддставяляя соответсвующие значения найдем вероятности (учтем что кол-во студентоы в группе С в 2 раза больше чем в других):

а)
$$P(B_1\:|\:A) = \frac{P(B_1) \cdot P(A\:|\:B_1)}{P(A)}= \frac{\frac{1}{4} \cdot 0.8}{\frac{1}{4} \cdot 0.8 + \frac{1}{4} \cdot 0.7 + \frac{1}{2} \cdot 0.9} \approx 0.24$$

б)
$$P(B_2\:|\:A) = \frac{P(B_2) \cdot P(A\:|\:B_2)}{P(A)}= \frac{\frac{1}{4} \cdot 0.7}{\frac{1}{4} \cdot 0.8 + \frac{1}{4} \cdot 0.7 + \frac{1}{2} \cdot 0.9} \approx 0.21$$

в)
$$P(B_3\:|\:A) = \frac{P(B_3) \cdot P(A\:|\:B_3)}{P(A)}= \frac{\frac{1}{2} \cdot 0.9}{\frac{1}{4} \cdot 0.8 + \frac{1}{4} \cdot 0.7 + \frac{1}{2} \cdot 0.9} \approx 0.55$$


Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

Решение:
    
а)  По формуле полной вероятности перемножим значения. 
$$ v=v1*v2*v3=0.1*0.2*0.25=0.005 $$ 
Т.е. вероятность равна 0.5%

б) 3 варианта, что сломаются 2 детали:

1) сломается первая и вторая деталь при этом не сломается третья
$$ v=v1*v2*\overline {v3} =0.1*0.2*0.75=0.015 $$ 

2) сломается первая и третья деталь при этом не сломается вторая

$$ v=v1*v3*\overline {v2} =0.1*0.25*0.8=0.02 $$ 

3) сломается вторяя и третья деталь при этом не сломается первая

$$ v=\overline {v1}*v2*v3 =0.9*0.2*0.25=0.045 $$ 

т.е. вероятность, что сломается 2 детали равна $0.015+0.02+0.045=0.08$

в) Посчитаем вероятность что не одна деталь не сломается:
$$ v=\overline {v1}*\overline {v2}*\overline {v3} =0.9*0.8*0.75=0.54 $$ 
Вероятность, что сломается хотя бы одна равна  $1- v=0.46$

г)
Значем вероятность что сломается хотя бы одна деталь $= 0.46$
Так же мы знаем выроятность, что сломается все три детали $= 0.005$

Т.е. вероятность того, что одна или две детали выйдут из строя равны $0.46-0.005=0.455$